msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/stress/AbstractMsalUiStressTest.java

```diff
@@ -0,0 +1,264 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.stress;
+
+import com.microsoft.identity.client.msal.automationapp.AbstractMsalUiTest;
+import com.microsoft.identity.client.msal.automationapp.R;
+import com.microsoft.identity.client.ui.automation.logging.appender.FileAppender;
+import com.microsoft.identity.client.ui.automation.logging.formatter.LogcatLikeFormatter;
+import com.microsoft.identity.client.ui.automation.performance.CPUMonitor;
+import com.microsoft.identity.client.ui.automation.performance.DeviceMonitor;
+import com.microsoft.identity.client.ui.automation.performance.MemoryMonitor;
+import com.microsoft.identity.client.ui.automation.performance.NetworkUsageMonitor;
+import com.microsoft.identity.client.ui.automation.performance.PerformanceProfile;
+import com.microsoft.identity.client.ui.automation.rules.RulesHelper;
+import com.microsoft.identity.client.ui.automation.utils.CommonUtils;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Before;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+
+public abstract class AbstractMsalUiStressTest<T, S> extends AbstractMsalUiTest {
+
+    // Sets the interval duration in seconds to which the device performance will be monitored.
+    private static final long DEVICE_MONITOR_INTERVAL = TimeUnit.SECONDS.toMillis(1);
+    private static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    private Exception executionException;
+
+    private int testsPassed = 0;
+    private int testsFailed = 0;
+
+    @Override
+    @Before
+    public void setup() {
+        super.setup();
+        executionException = null;
+
+        DeviceMonitor.setProfiler(PerformanceProfile.CPU, new CPUMonitor());
+        DeviceMonitor.setProfiler(PerformanceProfile.MEMORY, new MemoryMonitor());
+        DeviceMonitor.setProfiler(PerformanceProfile.NETWORK, new NetworkUsageMonitor());
+    }
+
+    /**
+     * Run the stress tests
+     */
+    public void run() throws Exception {
+        final T prerequisites = this.prepare();
+
+        final BlockingQueue<Runnable> blockingQueue = new ArrayBlockingQueue<>(getNumberOfThreads());
+        final RejectedExecutionHandler rejectedExecutionHandler = new ThreadPoolExecutor.CallerRunsPolicy();
+
+        final ExecutorService executorService = new ThreadPoolExecutor(
+                1,
+                getNumberOfThreads() + 1,
```

Comment:
Why +1?

Reply:
For the thread that generates the output file. It is kept alive the whole time when running the rest of the tests in the other threads.

Reply:
You're likely running more threads than you have cores.